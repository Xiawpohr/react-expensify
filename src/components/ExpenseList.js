import React from 'react'
import { connect } from 'react-redux'
import ExpenseListItem from './ExpenseListItem.js'
import getVisibleExpenses from '../selectors/getVisibleExpenses.js'
import Container from '../styles/Container.js'
import Text from '../styles/Text.js'
import { ListHeader, ListBody } from '../styles/List.js'

export const ExpenseList = (props) => (
  <Container>
    <ListHeader>
      <Text display={['block', 'none']} >Expenses</Text>
      <Text display={['none', 'block']} >Expense</Text>
      <Text display={['none', 'block']} >Amount</Text>
    </ListHeader>
    <ListBody>
      {
        !props.expenses || props.expenses.length === 0
          ? <ExpenseListItem />
          : props.expenses.map(expense => <ExpenseListItem key={expense.id} {...expense} />)
      }
    </ListBody>
  </Container>
)

const mapStateToProps = state => ({
  expenses: getVisibleExpenses(state.expenses, state.filters)
})

export default connect(mapStateToProps)(ExpenseList)
