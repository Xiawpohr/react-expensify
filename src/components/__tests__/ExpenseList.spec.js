import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseList } from '../ExpenseList.js'

const expenses = [{
  id: '1', description: 'water bill', amount: 100, createdAt: 2000
}, {
  id: '2', description: 'rent bill', amount: 2000, createdAt: 4200
}, {
  id: '3', description: 'electricity bill', amount: 500, createdAt: 3000
}]

test('should render ExpenseList with expenses correctly', () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />)
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpenseList without expenses', () => {
  const wrapper = shallow(<ExpenseList expenses={[]} />)
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpenseList without expenses props', () => {
  const wrapper = shallow(<ExpenseList />)
  expect(wrapper).toMatchSnapshot()
})
