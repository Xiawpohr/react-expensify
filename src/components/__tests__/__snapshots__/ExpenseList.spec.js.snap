// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses correctly 1`] = `
<Container>
  <ListHeader>
    <Text
      display={
        Array [
          "block",
          "none",
        ]
      }
    >
      Expenses
    </Text>
    <Text
      display={
        Array [
          "none",
          "block",
        ]
      }
    >
      Expense
    </Text>
    <Text
      display={
        Array [
          "none",
          "block",
        ]
      }
    >
      Amount
    </Text>
  </ListHeader>
  <ListBody>
    <ExpenseListItem
      amount={100}
      createdAt={2000}
      description="water bill"
      id="1"
      key="1"
    />
    <ExpenseListItem
      amount={2000}
      createdAt={4200}
      description="rent bill"
      id="2"
      key="2"
    />
    <ExpenseListItem
      amount={500}
      createdAt={3000}
      description="electricity bill"
      id="3"
      key="3"
    />
  </ListBody>
</Container>
`;

exports[`should render ExpenseList without expenses 1`] = `
<Container>
  <ListHeader>
    <Text
      display={
        Array [
          "block",
          "none",
        ]
      }
    >
      Expenses
    </Text>
    <Text
      display={
        Array [
          "none",
          "block",
        ]
      }
    >
      Expense
    </Text>
    <Text
      display={
        Array [
          "none",
          "block",
        ]
      }
    >
      Amount
    </Text>
  </ListHeader>
  <ListBody>
    <ExpenseListItem />
  </ListBody>
</Container>
`;

exports[`should render ExpenseList without expenses props 1`] = `
<Container>
  <ListHeader>
    <Text
      display={
        Array [
          "block",
          "none",
        ]
      }
    >
      Expenses
    </Text>
    <Text
      display={
        Array [
          "none",
          "block",
        ]
      }
    >
      Expense
    </Text>
    <Text
      display={
        Array [
          "none",
          "block",
        ]
      }
    >
      Amount
    </Text>
  </ListHeader>
  <ListBody>
    <ExpenseListItem />
  </ListBody>
</Container>
`;
